src/components/proposal_list/index.tsx

```diff
@@ -7,32 +6,21 @@ import {
   TableCell,
   TableRow,
 } from '@material-ui/core';
+import AccessTimeIcon from '@material-ui/icons/AccessTime';
 import { ProposalListProps } from './types';
 import { useGetStyles } from './styles';
 import {
   Title, Button, Status,
-} from './component';
-
-const Component = (props: { current: boolean, display: string }) => {
-  if (props.current === true) {
-    return (
-      <div className={classnames('button')}>
-        <Button display={props.display} />
-      </div>
-    );
-  }
-  return (
-    <div className={classnames('status')}>
-      <Status display={props.display} />
-    </div>
-  );
-};
+} from './components';
+import { ProposalUtils } from './utils';
 
 const ProposalList = (props: ProposalListProps) => {
   const {
-    data, className, desktop, imageUrl,
+    data, className, desktop, onClick,
   } = props;
+
   const { classes } = useGetStyles();
+  const { handleClick } = ProposalUtils(onClick);
```

Comment:
`ProposalUtils` should be `proposalUtils` or `useProposalUtils`

but side note: I was trying something when i moved it in to utils but I don't know how i feel about the structure and might stick with using this in hooks in the future.